pandajedi/jedicore/JediDBProxy.py

```diff
@@ -13622,49 +13623,52 @@ def updateInputFilesStagedAboutIdds_JEDI(self, jeditaskid, scope, filenames_dict
                         filenames_dict_without_ID[filename] = (datasetid, fileid)
                 # loop over files with fileID
                 if filenames_dict_with_fileID:
-                    varMaps = []
-                    for filename, (datasetid, fileid) in filenames_dict_with_fileID.items():
-                        tmp_varMap = varMap.copy()
-                        if scope != "pseudo_dataset":
-                            tmp_varMap[":lfn"] = filename
-                        else:
-                            tmp_varMap[":lfn"] = "%" + filename
-                        tmp_varMap[":fileID"] = fileid
-                        varMaps.append(tmp_varMap)
-                        tmpLog.debug(f"tmp_varMap: {tmp_varMap}")
-                    tmpLog.debug(f"running sql executemany: {sqlUF_with_fileID}")
-                    self.cur.executemany(sqlUF_with_fileID + comment, varMaps)
-                    retVal += self.cur.rowcount
+                    for one_batch in batched(filenames_dict_with_fileID.items(), chunk_size):
+                        # loop batches of executemany
+                        varMaps = []
+                        for filename, (datasetid, fileid) in one_batch:
+                            tmp_varMap = varMap.copy()
+                            tmp_varMap[":fileID"] = fileid
+                            del tmp_varMap[":scope"]
```

Comment:
it needs to check whether ':scope' is in the dict. when scope == "pseudo_dataset", ':scope' is not in the dict.
if ":scope" in tmp_varMap:
    del tmp_varMap[":scope"]